c/parallel/src/unique_by_key.cu

```diff
@@ -0,0 +1,514 @@
+  try
+  {
+    const char* name = "test";
+
+    const int cc      = cc_major * 10 + cc_minor;
+    const auto policy = unique_by_key::get_policy(cc);
+
+    const auto input_keys_it_value_t          = cccl_type_enum_to_name(input_keys_it.value_type.type);
+    const auto input_values_it_value_t        = cccl_type_enum_to_name(input_values_it.value_type.type);
+    const auto output_keys_it_value_t         = cccl_type_enum_to_name(output_keys_it.value_type.type);
+    const auto output_values_it_value_t       = cccl_type_enum_to_name(output_values_it.value_type.type);
+    const auto output_num_selected_it_value_t = cccl_type_enum_to_name(output_num_selected_it.value_type.type);
+    const auto offset_cpp                     = cccl_type_enum_to_name(cccl_type_enum::CCCL_INT64);
+    const cccl_type_info offset_t{sizeof(int64_t), alignof(int64_t), cccl_type_enum::CCCL_INT64};
+
+    const std::string input_keys_iterator_src = make_kernel_input_iterator(
+      offset_cpp,
+      get_iterator_name(input_keys_it, unique_by_key::unique_by_key_iterator_t::input_keys),
+      input_keys_it_value_t,
+      input_keys_it);
+    const std::string input_values_iterator_src = make_kernel_input_iterator(
+      offset_cpp,
+      get_iterator_name(input_values_it, unique_by_key::unique_by_key_iterator_t::input_values),
+      input_values_it_value_t,
+      input_values_it);
+    const std::string output_keys_iterator_src = make_kernel_output_iterator(
+      offset_cpp,
+      get_iterator_name(output_keys_it, unique_by_key::unique_by_key_iterator_t::output_keys),
+      output_keys_it_value_t,
+      output_keys_it);
+    const std::string output_values_iterator_src = make_kernel_output_iterator(
+      offset_cpp,
+      get_iterator_name(output_values_it, unique_by_key::unique_by_key_iterator_t::output_values),
+      output_values_it_value_t,
+      output_values_it);
+    const std::string output_num_selected_iterator_src = make_kernel_output_iterator(
+      offset_cpp,
+      get_iterator_name(output_num_selected_it, unique_by_key::unique_by_key_iterator_t::num_selected),
+      output_num_selected_it_value_t,
+      output_num_selected_it);
+
+    const std::string op_src = make_kernel_user_comparison_operator(input_keys_it_value_t, op);
+
+    constexpr std::string_view src_template = R"XXX(
+#include <cub/device/dispatch/kernels/scan.cuh>
+#include <cub/device/dispatch/kernels/unique_by_key.cuh>
+#include <cub/agent/single_pass_scan_operators.cuh>
+struct __align__({1}) storage_t {{
+  char data[{0}];
+}};
+struct __align__({3}) items_storage_t {{
+  char data[{2}];
+}};
+struct __align__({5}) num_out_storage_t {{
+  char data[{4}];
+}};
+{8}
+{9}
+{10}
+{11}
+{12}
+struct agent_policy_t {{
+  static constexpr int ITEMS_PER_THREAD = {7};
+  static constexpr int BLOCK_THREADS = {6};
+  static constexpr cub::BlockLoadAlgorithm LOAD_ALGORITHM = cub::BLOCK_LOAD_WARP_TRANSPOSE;
+  static constexpr cub::CacheLoadModifier LOAD_MODIFIER = cub::LOAD_LDG;
```

Comment:
suggestion: these two do not match the ones used in the tuning. Either use parameters in the dynamic policy or remove load algorithm and modifier from it.